kong/plugins/acme/schema.lua

```diff
@@ -248,6 +269,28 @@ local schema = {
         end
       }
     },
+    { custom_entity_check = {
+      field_sources = { "config.storage_config.redis.namespace", "config.storage_config.redis.scan_count", "config.storage_config.redis.auth", "config.storage_config.redis.ssl_server_name" },
+      fn = function(entity)
+        if (entity.config.storage_config.redis.namespace or ngx.null) ~= ngx.null then
+          deprecation("acme: config.storage_config.redis.namespace is deprecated, please use config.storage_config.redis.extra_options.namespace instead",
+            { after = "4.0", })
+        end
+        if (entity.config.storage_config.redis.scan_count or ngx.null) ~= ngx.null  then
```

Comment:
When a field has a `default` value, we also have to check against it, like we did here -> https://github.com/Kong/kong-ee/blob/76a8dc76b34c096fb49a4492d6ae7864636c5560/kong/plugins/datadog/schema.lua#L155